dashboard/app/models/user.rb

```diff
@@ -1033,7 +1033,9 @@ def should_see_edit_email_link?
 
   def should_see_add_password_form?
     !can_create_personal_login? && # mutually exclusive with personal login UI
-      can_edit_password? && encrypted_password.blank?
+      can_edit_password? && # allowed to edit password (i.e. not sponsored)
+      encrypted_password.blank? && # no password exists
+      !Policies::Lti.restricted_user?(self) # not restricted by their school district
```

Comment:
I considered adding the `!Policies::Lti.restricted_user?` check directly in the edit.html.haml (like I did below for the ManageLinkedAccounts section, however adding it to these methods that are called in the edit.html.haml means we can write unit tests.